tests/js/client/server_parameters/test-allow-collections-in-expressions-off.js

```diff
@@ -0,0 +1,132 @@
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB Inc, Cologne, Germany
+///
+/// @author Jan Steemann
+/// @author Copyright 2019, ArangoDB Inc, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+if (getOptions === true) {
+  return {
+    'query.allow-collections-in-expressions': "false",
+  };
+}
+const jsunity = require('jsunity');
+const errors = require('@arangodb').errors;
+const cn = "UnitTestsCollection";
+const en = "UnitTestsEdge";
+const db = require('internal').db;
+
+function testSuite() {
+  return {
+    setUpAll: function() {
+      db._drop(cn);
+      let c = db._create(cn);
+      let docs = [];
+      for (let i = 0; i < 500; ++i) {
+        docs.push({ _key: "test" + i, value: i });
+      }
+      c.insert(docs);
+      
+      db._drop(en);
+      db._createEdgeCollection(en);
+    },
+    
+    tearDownAll: function() {
+      db._drop(cn);
+      db._drop(en);
+    },
+
+    testUseInForLoop: function() {
+      let result = db._query("FOR doc IN " + cn + " RETURN doc").toArray();
+      assertEqual(500, result.length);
+      result.forEach((doc) => {
+        assertTrue(doc.hasOwnProperty('_key'));
+        assertTrue(doc.hasOwnProperty('value'));
+      });
+    },
+
+    testUseInSubquery: function() {
+      let result = db._query("LET sub = (FOR doc IN " + cn + " RETURN doc) RETURN sub").toArray();
+      assertEqual(1, result.length);
+      assertEqual(500, result[0].length);
+      result[0].forEach((doc) => {
+        assertTrue(doc.hasOwnProperty('_key'));
+        assertTrue(doc.hasOwnProperty('value'));
+      });
+    },
+    
+    testUseInInsert: function() {
+      let result = db._query("FOR i IN 1..1000 FILTER i > 10000 INSERT {} INTO " + cn).toArray();
+      assertEqual(0, result.length);
+    },
+    
+    testUseInUpdate: function() {
+      let result = db._query("FOR doc IN " + cn + " FILTER doc.value == 9999999 UPDATE doc WITH {} IN " + cn).toArray();
+      assertEqual(0, result.length);
+    },
+    
+    testUseInReplace: function() {
+      let result = db._query("FOR doc IN " + cn + " FILTER doc.value == 9999999 REPLACE doc WITH {} IN " + cn).toArray();
+      assertEqual(0, result.length);
+    },
+    
+    testUseInRemove: function() {
+      let result = db._query("FOR doc IN " + cn + " FILTER doc.value == 9999999 REMOVE doc IN " + cn).toArray();
+      assertEqual(0, result.length);
+    },
+    
+    testUseInTraversal: function() {
+      let result = db._query("WITH " + cn + " FOR v, e, p IN 1..3 OUTBOUND '" + cn + "/test0' " + en + " RETURN [v, e, p]").toArray();
+      assertEqual(0, result.length);
+    },
+    
+    testUseInShortestPath: function() {
+      let result = db._query("WITH " + cn + " FOR s IN OUTBOUND SHORTEST_PATH '" + cn + "/test0' TO '" + cn + "/test1' " + en + " RETURN s").toArray();
+      assertEqual(0, result.length);
+    },
+    
```

Comment:
I think there are some Calculations that explicitly want a collection by CollectionIdentifier not by name which will require at least a test here (maybe even some modification in the production code, as I expect those tests to fail)

What I found right now:
* `LENGTH(collection)` or `COLLECTION_COUNT(collection)` which are aliases of each other.
* The deprecated GeoIndexFunctions https://www.arangodb.com/docs/stable/aql/functions-geo.html#geo-index-functions (I am open to let them not work with the flag set)
* Fulltext functions: https://www.arangodb.com/docs/stable/aql/functions-fulltext.html#fulltext